src/pages/api/apiUtils/userData.ts

```diff
@@ -204,19 +216,31 @@ export const getPeriodGeoZoneTicketJson = async (
         periodExpiryAttributeInfo: ProductData | PeriodExpiryWithErrors,
     ): periodExpiryAttributeInfo is ProductData => (periodExpiryAttributeInfo as ProductData)?.products !== null;
 
-    const cookies = new Cookies(req, res);
-
     const nocCode = getNocFromIdToken(req, res);
+
+    const cookies = new Cookies(req, res);
     const periodTypeCookie = unescapeAndDecodeCookie(cookies, PERIOD_TYPE_COOKIE);
     const passengerTypeCookie = unescapeAndDecodeCookie(cookies, PASSENGER_TYPE_COOKIE);
     const operatorCookie = unescapeAndDecodeCookie(cookies, OPERATOR_COOKIE);
     const idToken = unescapeAndDecodeCookie(cookies, ID_TOKEN_COOKIE);
     const fareZoneCookie = unescapeAndDecodeCookie(cookies, CSV_ZONE_UPLOAD_COOKIE);
-    const periodExpiryAttributeInfo = getSessionAttribute(req, PERIOD_EXPIRY_ATTRIBUTE);
     const multipleProductCookie = unescapeAndDecodeCookie(cookies, MULTIPLE_PRODUCT_COOKIE);
 
-    if (!nocCode || !periodTypeCookie || !passengerTypeCookie || !operatorCookie || !idToken || !fareZoneCookie) {
-        throw new Error('Necessary session object or cookie not found to create user data json');
+    const periodExpiryAttributeInfo = getSessionAttribute(req, PERIOD_EXPIRY_ATTRIBUTE);
+    const timeRestriction = getSessionAttribute(req, TIME_RESTRICTIONS_DEFINITION_ATTRIBUTE);
+
+    if (
```

Comment:
could we potentially create a function with this as it is being used a few times in this file?

Reply:
It could do indeed. I've raised a ticket in the backlog to refactor this whole file because there are many parts which could be split out and reused.